### DISCUSSION TITLE
Wildcard in dimension name?

### DISCUSSION THREAD

**Discussion by AndrewILWilliams** [pydata/xarray#5803] [Q&A] [answered]
Hi there! 

I have a very weird dataset, where sometimes the `time` coordinate is labelled `time2` and sometimes is labelled `time3` (I don't know why).

Is there a way to use a wildcard in selecting this dimension? E.g. `da.mean("time*")`?

**Comment by max-sixty** [comment, marked as answer]
Hi @AndrewWilliams3142 !

There's no query functionality on var names at the moment.

It would be possible to rename them with a fairly simple function on the user side. 

In concept we could allow supplying a `re.match` object, though balanced against the larger API surface.

**Reply by AndrewILWilliams** [reply]
Thanks @max-sixty ! Yes I realised after I wrote this I could just query the `da.dims` and check like that. 

I suppose it might be tricky to allow this in xarray in case there are multiple dimensions that match the wildcard?

**Reply by ahuang11** [reply]
There's this: https://xarray.pydata.org/en/stable/generated/xarray.Dataset.filter_by_attrs.html

But pandas, you could filter columns by regex like
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.filter.html

So maybe xarray could also have something similar

**Comment by jthielen** [comment]
While not a fully generic wildcard handling, this specific issue with time coordinates (i.e., handling multiple coordinates of the same "physical type") is one of the motivating use cases for [cf-xarray](https://github.com/xarray-contrib/cf-xarray). With cf-xarray, this would look something like

```python
da.cf.mean("T")
```
